music21/features/jSymbolic.py

```diff
@@ -3559,9 +3571,10 @@ def process(self):
         s = self.data['partitionByInstrument']
         # each part has content for each instrument
         count = 0
-        for p in s.parts:
-            if p.recurse().notes:
-                count += 1
+        if s is not None:
+            for p in s.parts:
+                if p.recurse().notes:
+                    count += 1
```

Comment:
see above -- `partitionByInstrument` returns None if no instruments

I agree checking hasattr(s, 'parts') is unnecessary because the for loop won't execute

however -- I can reorder this so that short circuit happens first and we don't indent the following block

also -- this is an edge case where this is truly a count, so I'm torn on returning 0 vs. error

Reply:
whoops, no we definitely oughta check 'hasattr'!

Reply:
And returning 0 here doesn't seem very legitimate because we aren't searching everywhere in the stream to return a trustworthy zero, so I think raising is better.

Reply:
Actually, no, we don't need to check if it has parts because `partitionByInstrument` always returns parts, and I'm already checking if not s.

And if that behavior ever changes, an `AttributeError` will fail my tests that everything is a `FeatureException` so we should be good there